src/core/meshcop/joiner_router.cpp

```diff
@@ -141,15 +143,22 @@ ThreadError JoinerRouter::GetJoinerPort(uint16_t &aJoinerPort)
 
         if (tlv->GetType() == Tlv::kJoinerUdpPort)
         {
-            aJoinerPort = reinterpret_cast<JoinerUdpPortTlv *>(tlv)->GetUdpPort();
-            ExitNow(error = kThreadError_None);
+            return reinterpret_cast<JoinerUdpPortTlv *>(tlv)->GetUdpPort();
         }
 
         cur += sizeof(Tlv) + tlv->GetLength();
     }
 
 exit:
-    return error;
+    return OPENTHREAD_CONFIG_JOINER_UDP_PORT;
+}
+
+ThreadError JoinerRouter::SetJoinerUdpPort(uint16_t aJoinerUdpPort)
+{
+    mJoinerUdpPort = aJoinerUdpPort;
+    mIsJoinerPortConfigured = true;
+    mNetif.SetStateChangedFlags(OT_THREAD_NETDATA_UPDATED);
```

Comment:
Can we just call `HandleNetifStateChanged()` instead of triggering a network data update event?